client/src/main/java/com/paypal/selion/platform/grid/SeleniumGridListener.java

```diff
@@ -137,10 +137,13 @@ public void beforeInvocation(IInvokedMethod method, ITestResult testResult) {
                     System.exit(1); // NOSONAR
                 }
             }
+        } catch (RuntimeException e) { // NOSONAR
+            Grid.getThreadLocalException().set(e);
         } catch (Exception e) { // NOSONAR
-            logger.log(Level.WARNING, "An error occurred while processing beforeInvocation: " + e.getMessage(), e);
+            //Must wrap the Exception to RuntimeException, So that users no need to
+            //add try catch block in their test case when invoke Grid.driver()
+            Grid.getThreadLocalException().set(new RuntimeException(e.getMessage(), e));
```

Comment:
Why not convert to RuntimeException at the throw site instead?
   
Also what if multiple exceptions have occurred in BeforeX config methods before Grid.driver().  We are saving only the last and clobbering all earlier ones.